pkg/daemon/multus/resources.go

```diff
@@ -155,50 +189,62 @@ func (vt *ValidationTest) startClients(
 	return nil
 }
 
-// if this returns success, the return value is guaranteed to be greater than 0
-func (vt *ValidationTest) getExpectedNumberOfClientPods(ctx context.Context) (int, error) {
+func (vt *ValidationTest) getExpectedNumberOfDaemonSetPods(
+	ctx context.Context,
+	daemonsetSelectorLabel string,
+	expectedNumDaemonsets int,
+) (int, error) {
 	var agreedNumberScheduled int32
 	listOpts := meta.ListOptions{
-		LabelSelector: clientAppLabel(),
+		LabelSelector: daemonsetSelectorLabel,
 	}
 	dsets, err := vt.Clientset.AppsV1().DaemonSets(vt.Namespace).List(ctx, listOpts)
 	if err != nil {
-		return 0, fmt.Errorf("unexpected error listing client daemonsets: %w", err)
+		return 0, fmt.Errorf("unexpected error listing daemonsets: %w", err)
 	}
-	if len(dsets.Items) < vt.DaemonsPerNode {
-		return 0, fmt.Errorf("got fewer client daemonsets [%d] than should exist [%d]", len(dsets.Items), vt.DaemonsPerNode)
+	if len(dsets.Items) != expectedNumDaemonsets {
+		return 0, fmt.Errorf("got %d daemonsets when %d should exist", len(dsets.Items), expectedNumDaemonsets)
 	}
 
 	for _, d := range dsets.Items {
 		numScheduled := d.Status.CurrentNumberScheduled
 		if numScheduled == 0 {
-			return 0, fmt.Errorf("a daemonset expects zero scheduled client pods; cannot validate with no running clients")
+			return 0, fmt.Errorf("a daemonset expects zero scheduled pods")
 		}
 		if agreedNumberScheduled == 0 {
 			agreedNumberScheduled = numScheduled
 		} else if numScheduled != agreedNumberScheduled {
-			return 0, fmt.Errorf("daemonsets do not all agree on the number of expected client pods")
+			return 0, fmt.Errorf("daemonsets do not all agree on the number of expected pods")
 		}
 	}
 
-	return int(agreedNumberScheduled) * vt.DaemonsPerNode, nil
+	return int(agreedNumberScheduled) * expectedNumDaemonsets, nil
 }
 
-func (vt *ValidationTest) allClientsAreRunning(ctx context.Context, expectedNumPods int) (bool, error) {
-	pods, err := vt.getClientPods(ctx, expectedNumPods)
+func (vt *ValidationTest) getNumRunningPods(
+	ctx context.Context,
+	podSelectorLabel string,
+	expectedNumTotalPods int,
+) (int, error) {
```

Comment:
Replace check that all clients are running with re-usable and more flexible version that just returns the number of pods.